File: i2f-secure/src/main/java/i2f/springboot/secure/SecureConfig.java
package i2f.springboot.secure;


import i2f.springboot.secure.consts.SecureConsts;
import i2f.springboot.secure.core.SecureTransferFilter;
import i2f.springboot.secure.data.SecureCtrl;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.web.servlet.FilterRegistrationBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.servlet.DispatcherType;
import java.util.Set;

/**
 * @author ltb
 * @date 2022/6/29 15:50
 * @desc
 */
@ConditionalOnExpression("${i2f.springboot.config.secure.enable:true}")
@Data
@Configuration
@ConfigurationProperties(prefix = "i2f.springboot.config.secure")
public class SecureConfig {
    private String rsaStorePath = "../";

    private String responseCharset = "UTF-8";

    private int rsaKeySize = 1024;

    private int randomKeyBound = 8192;

    private int nonceTimeoutSeconds = 6 * 60;

    private boolean enableDynamicRsaKey = true;

    private long dynamicRefreshDelaySeconds = 6 * 60;

    private int dynamicMaxHistoriesCount = 5;

    private String headerName = SecureConsts.DEFAULT_SECURE_HEADER_NAME;

    private String headerSeparator = SecureConsts.DEFAULT_HEADER_SEPARATOR;

    private String dynamicKeyHeaderName = SecureConsts.SECURE_DYNAMIC_KEY_HEADER;

    private String encUrlPath = SecureConsts.DEFAULT_ENC_URL_PATH;

    private String parameterName=SecureConsts.DEFAULT_SECURE_PARAMETER_NAME;

    private SecureCtrl defaultControl = new SecureCtrl(true, true);

    private SecureWhiteListConfig whiteList;

    @Data
    @NoArgsConstructor
    public static class SecureWhiteListConfig {
        private Set<String> bothPattens;
        private Set<String> inPattens;
        private Set<String> outPattens;
    }

    @Autowired
    private AutowireCapableBeanFactory autowireCapableBeanFactory;

    @Bean
    public FilterRegistrationBean<SecureTransferFilter> filterFilterRegistrationBean() {
        SecureTransferFilter filter = new SecureTransferFilter();
        autowireCapableBeanFactory.autowireBean(filter);
        FilterRegistrationBean<SecureTransferFilter> registrationBean = new FilterRegistrationBean<>();
        registrationBean.setFilter(filter);
        registrationBean.addUrlPatterns("/**");
        registrationBean.setOrder(-1);
        registrationBean.setDispatcherTypes(DispatcherType.REQUEST, DispatcherType.FORWARD);
        return registrationBean;
    }
}

File: i2f-secure/web/secure-web/src/secure/secure-config.js
/**
 * 主配置
 */
import SecureConsts from "./consts/secure-consts";

const SecureConfig={
   // AES秘钥长度，默认128，可选128,192,256
   aesKeySize: SecureConsts.AES_KEY_SIZE_128(),
   // 随机秘钥生成的随机数的最大值，默认8192
   randomKeyBound: 8192,
   // 用于存储安全头的请求头名称，默认sswh
   headerName: SecureConsts.DEFAULT_SECURE_HEADER_NAME(),
   // 动态刷新RSA秘钥的响应头，默认skey
   dynamicKeyHeaderName: SecureConsts.SECURE_DYNAMIC_KEY_HEADER(),
   // 安全头格式的分隔符，默认;
   headerSeparator: SecureConsts.DEFAULT_HEADER_SEPARATOR(),
   // 指定在使用编码URL转发时的转发路径
   encUrlPath: SecureConsts.ENC_URL_PATH(),
   // 安全URL参数的参数名称
   parameterName:SecureConsts.DEFAULT_SECURE_PARAMETER_NAME(),
   // 是否开启详细日志
   // 在正式环境中，请禁用
   enableDebugLog: true,
   // 加密配置的白名单url
   whileList:['/secure/key','/login'],
   // 加密URL的URL白名单
   encWhiteList:['/secure/key','/login']
}

export default SecureConfig